pkgs/build-support/mkshell/default.nix

```
@@ -41,10 +41,15 @@ stdenv.mkDerivation ({
   shellHook = lib.concatStringsSep "\n" (lib.catAttrs "shellHook"
     (lib.reverseList inputsFrom ++ [ attrs ]));
 
-  nobuildPhase = ''
-    echo
-    echo "This derivation is not meant to be built, aborting";
-    echo
-    exit 1
+  phases = [ "buildPhase" ];
+
+  buildPhase = ''
+    echo "------------------------------------------------------------" >>$out
+    echo " WARNING: this file existence or location is not guaranteed." >>$out
```

WDYT of something like:

```suggestion
    echo " WARNING: the existence of this path is not guaranteed." >>$out
```

---

The warning is already scary enough with your changes so we don't have to add even more text.